driver/tke_driver.go

```diff
@@ -621,129 +651,110 @@ func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, opts *type
 		return nil, err
 	}
 
-	logrus.Debugf("Updating config, clusterName: %s, clusterVersion: %s, new node: %v", state.ClusterName, state.ClusterVersion, newState.GoodsNum)
+	logrus.Debugf("Updating config, clusterName: %s, clusterVersion: %s, new node: %v", *state.CreateClusterRequest.ClusterBasicSettings.ClusterName, *state.CreateClusterRequest.ClusterBasicSettings.ClusterVersion, newState.NodeCount)
 
 	if state.NodeCount != newState.NodeCount {
-		request := ccs.NewDescribeClusterInstancesRequest()
-		request.ClusterID = state.ClusterID
-		resp, err := svc.DescribeClusterInstance(request)
+		request := tke.NewDescribeClusterInstancesRequest()
+		request.ClusterId = &state.ClusterID
+		resp, err := svc.DescribeClusterInstances(request)
 		if _, ok := err.(*tcerrors.TencentCloudSDKError); ok {
 			return nil, fmt.Errorf("an API error has returned: %s", err)
 		}
-		nodeCount := resp.Data.TotalCount
+		nodeCount := resp.Response.TotalCount
 
-		if newState.NodeCount > nodeCount {
+		if newState.NodeCount > int64(*nodeCount) {
 
 			log.Infof(ctx, "Scaling up cluster nodes to %d", newState.NodeCount)
-			req, err := getWrapAddClusterInstancesRequest(state, newState, nodeCount)
+			req, err := getWrapAddClusterInstancesRequest(state, newState, int64(*nodeCount))
 			if err != nil {
 				return nil, err
 			}
 
-			resp, err := svc.AddClusterInstances(req)
+			_, err = svc.CreateClusterInstances(req)
 			if _, ok := err.(*tcerrors.TencentCloudSDKError); ok {
 				return nil, err
 			}
 			if err == nil {
-				logrus.Infof("Add cluster instances is called for cluster %s. Status Code %v, Response Instance IDs: %s", state.ClusterID, resp.Code, resp.Data.InstanceIDs)
+				logrus.Infof("Add cluster instances is called for cluster %s.", state.ClusterID)
 			}
 			if err := waitTKECluster(ctx, svc, state); err != nil {
 				return nil, err
 			}
-		} else if newState.NodeCount < resp.Data.TotalCount {
+		} else if newState.NodeCount < int64(*nodeCount) {
 			log.Infof(ctx, "Scaling down cluster nodes to %d", newState.NodeCount)
-			if err := removeClusterInstances(state, newState, svc, resp); err != nil {
+
+			req, err := removeClusterInstances(state, newState, int64(*nodeCount), resp)
+			if err != nil {
+				return nil, err
+			}
+
+			_, err = svc.DeleteClusterInstances(req)
+			if _, ok := err.(*tcerrors.TencentCloudSDKError); ok {
```

Comment:
What if the error is a mismatching type, looks like we don't handle it in this case.